backend/mem/fileSystem.go

```diff
@@ -0,0 +1,78 @@
+package mem
+
+import (
+	"github.com/c2fo/vfs/v4"
+	"github.com/c2fo/vfs/v4/backend"
+	"github.com/c2fo/vfs/v4/utils"
+	"path"
+)
+
+//Scheme defines the filesystem type.
+const Scheme = "mem"
+const name = "In-Memory Filesystem"
+
+var systemMap map[string]*File
+var fileList []*File
```

Comment:
Rather than making these global, can we make them a part of the FileSystem struct?